prov/gni/src/gnix_msg.c

```diff
@@ -1888,6 +1894,48 @@ static int __smsg_eager_msg_w_data(void *data, void *msg)
 	req = _gnix_match_tag(posted_queue, hdr->msg_tag, 0, FI_PEEK, NULL,
 			      &vc->peer_addr);
 	if (req) {
+		if (req->type == GNIX_FAB_RQ_MRECV) {
+
+			mrecv_req = req;
+			req = _gnix_fr_alloc(ep);
+			if (req == NULL) {
+				return -FI_ENOMEM;
+			}
+
+			/*
+			 * set recv related fields in request,
+			 * and adjust mrecv buffer pointer and
+			 * space remaining
+			 */
+
+			req->type = GNIX_FAB_RQ_RECV;
+			req->msg.recv_flags = mrecv_req->msg.recv_flags;
+			req->msg.recv_flags |= FI_MULTI_RECV;
+			req->msg.recv_info[0].recv_addr =
+				mrecv_req->msg.mr_buf_addr;
+			req->msg.recv_info[0].recv_len =
+				mrecv_req->msg.mr_space_left;
+			req->msg.cum_recv_len = mrecv_req->msg.mr_space_left;
+			req->user_context = mrecv_req->user_context;
+
+			req->msg.parent = mrecv_req;
```

Comment:
in what way?  The mother is the multi recv request.  The children are all the "regular" receive requests that are being used to progress the actual receives.